### PR TITLE
Feature/story data modelling

### PR DESCRIPTION
**Pull request by bramirez96**
# Description

Implementing router, testing, and documentation for access to story data from the frontend.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by JLC6290 — APPROVED** [review]
Everything looks clean, Middleware is well documented.

File: api/middleware/storyValidation.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * A custom middleware that checks to ensure the data passed in is valid before
+ * attempting to send it to the database. This allows for better error messages
+ * to be sent back to the client by resolving the API call to a 400 if the data
+ * is incorrectly formatted. If the data is correct, the server will move on to
+ * the actual POST router.
+ * @param {Object} req the server request object
+ * @param {Object} res the server response object
+ * @param {Function} next a function that will continue to the next middleware
+ */
+const storyValidation = (req, res, next) => {
```
Comment (Schrese) [inline api/middleware/storyValidation.js:13]: Great idea to validate this!

File: api/stories/storyRouter.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - Stories
+ *    parameters:
+ *      - $ref: '#/components/parameters/storyId'
+ *    responses:
+ *      204:
+ *        $ref: '#/components/responses/EmptySuccess'
+ *      401:
```
Comment (Schrese) [inline api/stories/storyRouter.js:236]: Just a question. Where's the 401 coming from?

**Review by Schrese — APPROVED** [review]
Looks great! 
I also like the use of lodash

**Review by Nick-Ohman — APPROVED** [review]
This looks great, good job!

**Review by SandraCoburn — APPROVED** [review]
Good job Brandon!

**Pull request merged**